spock-core/src/main/java/org/spockframework/runtime/extension/ParameterResolver.java

```diff
@@ -0,0 +1,39 @@
+package org.spockframework.runtime.extension;
+
+import org.spockframework.runtime.model.ParameterInfo;
+import org.spockframework.util.Assert;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.spockframework.runtime.model.MethodInfo.MISSING_ARGUMENT;
+
+/**
+ * Helper class for resolving parameters.
+ */
+public class ParameterResolver {
+  public static void resolveParameter(IMethodInvocation invocation, ParameterInfo parameterInfo, Supplier<Object> valueSupplier) {
+    int index = parameterInfo.getIndex();
+    Assert.that(invocation.getArguments()[index] == MISSING_ARGUMENT, "Parameter " + parameterInfo.getName() + " is already set");
+    invocation.getArguments()[index] = valueSupplier.get();
+  }
+
+  /**
+   * Convenience interceptor that resolves a parameter.
+   */
+  public static class Interceptor implements IMethodInterceptor {
+    private final ParameterInfo parameterInfo;
+    private final Function<IMethodInvocation, Object> valueSupplier;
```

Comment:
Do you think it makes sense to have the `Interceptor` typed (`Interceptor<RV>`) to use as a type of the value supplier?

I might imagine, some extension will only support a specified type of parameters (to be filtered out before the instance of that Interceptor is created). For generic cases, `Object` could be used after all.

Reply:
The generic type info wouldn't be available at runtime for any checks, so what would we get from making it generic? The argument array is of type `Object[]` so no need for a generic type as well on this side.

Reply:
However, it was just an idea. It might not have a big impact in reality. Feel free to skip it.